File: mns-rest/src/test/java/cz/fb/manaus/rest/MetricsControllerTest.java
package cz.fb.manaus.rest;

import com.codahale.metrics.MetricRegistry;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;
import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;


public class MetricsControllerTest extends AbstractControllerTest {

    @Autowired
    private MetricRegistry metricRegistry;

    @Test
    public void testMetrics() throws Exception {
        checkResponse("/metrics", "metric.get", "1");
    }

    @Test
    public void testUpdateHistogram() throws Exception {
        mvc.perform(post(
                "/metrics/histogram/{name}", "histogram.test")
                .contentType(MediaType.APPLICATION_JSON)
                .content("100"))
                .andExpect(status().isOk())
                .andReturn();
        assertThat(metricRegistry.histogram("histogram.test").getCount(), is(1L));
    }

    @Test
    public void testUpdateMeter() throws Exception {
        mvc.perform(post(
                "/metrics/meter/{name}", "meter.test")
                .contentType(MediaType.APPLICATION_JSON)
                .content("100"))
                .andExpect(status().isOk())
                .andReturn();
        assertThat(metricRegistry.meter("meter.test").getCount(), is(100L));
    }

    @Test
    public void testUpdateCounter() throws Exception {
        mvc.perform(post(
                "/metrics/counter/{name}", "counter.test")
                .contentType(MediaType.APPLICATION_JSON)
                .content("100"))
                .andExpect(status().isOk())
                .andReturn();
        assertThat(metricRegistry.counter("counter.test").getCount(), is(100L));
    }
}

File: mns-rest/src/test/java/cz/fb/manaus/rest/MetricsControllerTest.kt
package cz.fb.manaus.rest

import com.codahale.metrics.MetricRegistry
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.MediaType
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
import kotlin.test.assertEquals


class MetricsControllerTest : AbstractControllerTest() {

    @Autowired
    private lateinit var metricRegistry: MetricRegistry

    @Test
    fun testMetrics() {
        checkResponse("/metrics", "metric.get", "1")
    }

    @Test
    fun `update histogram`() {
        mvc.perform(post(
                "/metrics/histogram/{name}", "histogram.test")
                .contentType(MediaType.APPLICATION_JSON)
                .content("100"))
                .andExpect(status().isOk)
                .andReturn()
        assertEquals(1, metricRegistry.histogram("histogram.test").count)
    }

    @Test
    fun `update meter`() {
        mvc.perform(post(
                "/metrics/meter/{name}", "meter.test")
                .contentType(MediaType.APPLICATION_JSON)
                .content("100"))
                .andExpect(status().isOk)
                .andReturn()
        assertEquals(100, metricRegistry.meter("meter.test").count)
    }

    @Test
    fun `update counter`() {
        mvc.perform(post(
                "/metrics/counter/{name}", "counter.test")
                .contentType(MediaType.APPLICATION_JSON)
                .content("100"))
                .andExpect(status().isOk)
                .andReturn()
        assertEquals(100, metricRegistry.counter("counter.test").count)
    }
}